### DISCUSSION TITLE
Is there any migration guide to update from YARP v2?

### DISCUSSION THREAD

**Discussion by owocado** [robotology/yarp#3134] [Support - Q&A] [answered]
Hello community,

I am currently using very old version of YARP v2.3.72 in my 5 year old C++ project. I am in the process of updating it to latest YARP version and I was reading release notes for each release since 2.3.72 carefully to see what breaking changes I need to account for but it start to feel exhausting and each release note is HUGE.

I am only using `yarp::os` and `yarp::sig` libraries. I was hoping if there is a official/unofficial migration guide that gives summary of all the breaking changes between 2.3.72 to 3.9.0, sort of TLDR maybe, otherwise I can do it the way what I'm doing - going through each release note one by one and go from there which would take weeks. I'm only trying to save some time if possible. 😄 

If anyone can give me guidance/tips on this, it would be really appreaciated. Thanks for reading and wish you a great day ahead!

**Comment by PeterBowman** [comment, marked as answer]
Hello, @owocado. Back in the day our org roboticslab-uc3m followed the migration path detailed in https://github.com/roboticslab-uc3m/questions-and-answers/issues/83. There are several details specific to us, but perhaps it could be of use for you as well. Note it only covers the transition from 2.x to 3.0.

**Reply by owocado** [reply]
Oh this is really helpful resource to get myself to 3.0, thank you for sharing this. Once I transition my old code to 3.0, I can slowly update it to support latest YARP release from there.

Thank you very much for taking the time to respond promptly and wish you a great day ahead!
